File: QytetetJava/src/modeloqytetet/OtraCasilla.java
/**
 * Practica de:
 * 
 * Francisco Beltrán Sánchez
 * Pablo Cordero Romero
 */
package modeloqytetet;

public class OtraCasilla extends Casilla{
    TipoCasilla tipo;

    OtraCasilla(int numeroCasilla, int coste, TipoCasilla tp){
        super(numeroCasilla, coste);
        this.tipo = tp;
    }

    @Override
    protected TipoCasilla getTipo(){
        return tipo;
    }

    @Override
    protected boolean soyEdificable(){
        return false;
    }
    
    @Override
    protected TituloPropiedad getTitulo() {
        return null;
    }

    @Override
    public boolean tengoPropietario(){
        return false;
    }
    
    @Override
    public String toString() {
        return super.toString() + "\n\ttipo=" + tipo;
    }
}

File: QytetetRuby/lib/otraCasilla.rb
#encoding: utf-8
# Francisco Beltrán Sánchez
# Pablo Cordero Romero
module ModeloQytetet
    class OtraCasilla < Casilla
        attr_reader :tipo
        attr_reader :titulo

        def initialize(numCas, coste, tp)
            super(numCas, coste)
            @tipo = tp
            @titulo = nil
        end

#        def self.copia(casilla, tipo)
#            super(casilla)
#            @tipo = tipo
#          
#            return self
#        end

        def soyEdificable
            salida = false
            salida
        end

        def tengoPropietario
            salida = false
            salida
        end 

        def to_s
          "#{super}\n\tTipo: #{@tipo}"
        end
    end
end